storage/diemdb/src/transaction_store/mod.rs

```diff
@@ -168,6 +168,13 @@ impl TransactionStore {
             .ok_or_else(|| DiemDbError::NotFound(format!("WriteSet at version {}", version)).into())
     }
 
+    /// Get the first version that write set starts existent.
+    pub fn get_first_write_set_version(&self) -> Result<Option<Version>> {
+        let mut iter = self.db.iter::<WriteSetSchema>(Default::default())?;
```

Comment:
Wooo! 🚀 @lightmark, do you think we could also add one for the transactions or transactions info schema? So we can tell where the first transaction is stored

Reply:
Also: silly question. How does `Default::default()` differ from `ReadOptions::default()`?

Reply:
For default, there is no difference. I guess using `Default` trait is pretty new and seems more elegant since it can be derived.

For txn and output, because we don't prune any txn/output now, there is no such api. I prefer to implement it once we started to prune it otherwise it seems a little bit over-engineering, what do you think? cc: @msmouse

Reply:
@lightmark, when state sync v2 lands (at the end of this year), nodes might start up and do the following: (i) download all account states at a version `V` (where `V > 0`); and then (ii) start to execute transaction and/or apply transaction outputs from `V`. As a result, not all nodes will store all transactions/transaction infos since genesis. So, this is a problem we already have to solve today (even without pruning). Thoughts?

Reply:
Are we going to keep txn and txn_info in sync? if yes, we will keep only one api for that, right? cuz if we implement this api for each piece of data, it admits the possibility of being out of sync indirectly. @msmouse thoughts?

Reply:
@lightmark, yes -- we're going to keep transactions and transaction infos together (in sync). Transaction outputs are "optional". We'll only need one API to fetch the transaction infos 😄

Reply:
Yeah one API for both would be fine.

And as discussed elsewhere, breaking the "complete ledger history assumption" might introduce other stuff we'd need to address. But I'm not too hesitate to start building APIs around it (like returning the oldest txn version we have).

Reply:
I'll cover that as your networking buddy is going to abandon you...lol

Reply:
Haha, thanks @lightmark and @msmouse 😄